### PR TITLE
Use a newer version of pynYNAB api.

### PR DESCRIPTION
**Pull request by rossdargan**
Please check to see if you are happy for the requirments.txt to be the way they are!

### CODE REVIEW

File: python/main.py [hunk 1]
```diff
@@ -85,13 +93,8 @@ def getpayee(payeename):
...
         )
 
-        if not ynab_client.budget.be_transactions.containsduplicate(transaction):
-            log.debug('Appending transaction %s '%transaction.getdict())
-            transactions.append(transaction)
-        else:
-            log.debug('Duplicate transaction found %s '%transaction.getdict())
-
-        ynab_client.add_transactions(transactions)
+        ynab_client.budget.be_transactions.append(transaction)
```
Comment (scottrobertson) [inline python/main.py]: @rossdargan sorry for the delay. Are we ok with getting rid of duplicate checking here? We needed it for webhook retrying.

**Comment by ganey** [comment]
I've been getting duplicates from nearly every webhook recently

**Comment by rossdargan** [comment]
Yeah me to... My understanding is ynab should only send duplicates if the
API call fails... My guess is it's timing out on there end which is why the
dup check is still needed. Guess it needs adding back.

My bad!

**Comment by scottrobertson** [comment]
@rossdargan you ok taking care of that, or would you like me to?

**Comment by rossdargan** [comment]
@scottrobertson I'll take a look. It might be a few days though before I sit in front of my home PC, and get a chance to test it!

**Comment by scottrobertson** [comment]
@rossdargan no problem. I will try and look tonight. Gonna get this tested and merged too (even without that fix)

**Comment by rossdargan** [comment]
Easy tiger - I'm about to comitt a fix lol

**Comment by scottrobertson** [comment]
@rossdargan struggling to get this working actually.

```
DEBUG:main:server_knowledge_of_device 0
DEBUG:main:current_server_knowledge 519727589
DEBUG:main:Server knowledge has gone up by 519727589. We should be getting back some entities from the server
DEBUG:main:current_device_knowledge 0
DEBUG:main:device_knowledge_of_server 519727589
ERROR:main:Exception on /webhook [POST]
Traceback (most recent call last):
  File "./monzo-to-ynab/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "./monzo-to-ynab/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "./monzo-to-ynab/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "./monzo-to-ynab/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "./monzo-to-ynab/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "./monzo-to-ynab/python/main.py", line 41, in route_webhook
    ynab_client.sync()
  File "./monzo-to-ynab/src/pynynab/pynYNAB/Client.py", line 132, in sync
    self.select_budget(self.budget_name)
  File "./monzo-to-ynab/src/pynynab/pynYNAB/Client.py", line 236, in select_budget
    raise BudgetNotFound()
BudgetNotFound
```

But the budget is 100% there...

**Comment by scottrobertson** [comment]
Never mind, i see that the library uses `be_budget_versions` now, and not `be_budgets`

**Comment by scottrobertson** [comment]
Wow this is painfully slow for bigger budgets :(

**Comment by rossdargan** [comment]
Yeah... I wonder if we need to load the budget when the app starts and just do a sync on each call.

The reason we get duplicates is because it's timing out on the web hook 👎

**Comment by scottrobertson** [comment]
I have increased the timeout for now as a quick fix

**Comment by scottrobertson** [comment]
Do you mind if i merge this, and you can do the fix in another PR?

**Comment by rossdargan** [comment]
Sure, Go ahead. I have had another change downstream from someone else to merge in anyway.

**Comment by rossdargan** [comment]
I've added in what should have been a dup check... but it isn't working. I've checked it in if you want to see what I'm doing: https://github.com/rossdargan/monzo-to-ynab/blob/master/python/main.py

**Pull request merged**